### ISSUE TITLE
Create "Submit a Game" Form

### ISSUE THREAD

**Issue by foohonpie** [pbbg/pbbg.com#12]
Current site for reference:

<img width="778" alt="Screen Shot 2020-10-02 at 4 19 54 PM" src="https://user-images.githubusercontent.com/674157/94976852-2163e880-04cb-11eb-9838-85fa7e611f92.png">

I'll update this ticket when the /games POST endpoint is ready, but we can get started on the UI.  Some notes:

* URL can be /games/create like the current site, unless we think something else is better (/add? /new?).
* Can ignore the "Get info" button for now, I'll make a separate ticket for that.
* Signed-in user isn't required for now.
* Success can go to index for now.

**Comment by bpkennedy** [comment]
AC (Acceptance Criteria) questions below!

* Modality - is this a modal with darkened overlay on desktop or is it full screen?  Is it the same behavior on desktop vs mobile?  I assume it is "routable" directly as you mentioned the /games/create path.  (FYI modals can be routes as well as full screen in Quasar/Vue).
* Responsive UI - for better display on mobile I suggest not aligning input labels horizontally to the inputs themselves and instead stacking (or using the interior placeholder, a la Material Design style that Quasar has). 
* Navigation - how does a user "go back" or leave this without submitting?
* Form validation - which fields are required?  What validation rules (max chars on name, url, short description?). Are fields case sensitive?  Should url field try to do any validation on/before Submit (like check for 200 http status or something?).
* Image field - what are allowed file types? What is max file size?  Recommended aspect ratio/size?  Cropping required? (this would probably be a fair bit of extra work as I don't think Quasar has a cropping solution built in).
* User Experience - on click of Submit do we show a brief loader inside of the submit button, inside the panel with an overlay, or over the entire viewport?  After the submission is successful can we show a "toast" message in the lower left of the viewport confirming that the Game has been created/saved/submitted?  If the submission fails, can we show a toast with a failure message or some other kind of alert somewhere?

**Comment by bpkennedy** [comment]
* Validation (cont'd): across the app will we prefer form field validation to pop/show error messages as each field is blurred away from or altogether only on the submit action?  And after error/validation messages are showing and the user is fixing them - same question...do they "resolve" on blur of each field or stay showing until the user clicks submit again?

**Comment by foohonpie** [comment]
Great questions!

> Modality

Let's make it non-modal.  There will likely be lots of white space, and we'll probably refactor/redesign at some point, but we can start with just the form on the page given that we don't have other content beneath it.

> Responsive UI

I defer to your judgement regarding changes to the form layout.  The current site is for reference, but we can/should modify the design where it makes sense.

> Navigation

This submission link is often shared directly, so there isn't a clear place to go back to.  The user should be able to get back with the general nav, which will need updating to include at least a link to index.  We can sort that out on a different ticket.

> Form validation

This is pending some decisions on the backend, but to start we can go with a required check on:

* url
* name
* short_description

> Image field

This is also pending some backend stuff.  The cropping may (should?) be something that we pass to the backend to handle entirely.  For this ticket we can leave off FE restrictions, and I'll use it to test against the backend validation.

> User Experience

No preference on loading states or notifications.

> Validation (cont'd)

I'd prefer the FE to validate in place so the user can resolve any issues before pressing submit.  However, we should also anticipate additional validation from the backend coming all at once.  Details pending.

---

I anticipate that we'll be iterating given that the API is being developed in parallel.  This ticket can be for a functional phase one knowing that we'll come back around once we've put more thought into larger redesign considerations.

**Comment by foohonpie** [comment]
Let's wait on the image handling; I'd like to pull down home page screenshots instead for the initial data load.  We can assess the needs there as part of the "get info" work (#13).

**Comment by foohonpie** [comment]
This just got an update via #14, but is blocked by https://github.com/pbbg/api.pbbg.com/issues/5.
